rust/core-lib/src/editor.rs

```diff
@@ -961,27 +961,13 @@ impl Editor {
         self.view.collapse_selections(&self.text);
     }
 
-    fn make_uppercase(&mut self) {
+    fn transform_text(&mut self, transform_function: &Fn(String) -> String) {
```

Comment:
cmyr, should the Fn accept a &String?  Not sure what as written implies about the source string being copied.  Also, should this be an FnMut instead?

Reply:
I'm going to hold off on this until tomorrow when I'm bright eyed and caffeinated, because I don't want to suggest something daft and then overrule myself in the morning.